internal/consumer/service_test.go

```diff
@@ -0,0 +1,45 @@
+package consumer
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/mock"
+)
+
+type DbRepoMock struct {
+	mock.Mock
+}
+
+func (m *DbRepoMock) SaveItem(item Item) error {
+	m.Called()
+	return nil
+}
+
+type DataServiceMock struct {
+	mock.Mock
+}
+
+func (m *DataServiceMock) getTopStories() ([]int, error) {
+	return []int{1, 2, 3, 4, 5}, nil
+}
+
+func (m *DataServiceMock) getItem(id int) (*Item, error) {
+	item := Item{
+		Deleted: false,
+		Dead:    false,
+	}
+
+	return &item, nil
+}
+
+func TestConsumer_Execute(t *testing.T) {
```

Comment:
Are you adding tests for the other parts later on? :)

Reply:
i have added a few more tests and now using table testing